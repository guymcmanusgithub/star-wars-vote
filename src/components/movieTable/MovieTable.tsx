import React from 'react'
import { Table, Caption, THead, TBody,Tr,Th, Td, Button } from './MovieTable.style'

export interface MovieTableProps {
  movieInfo: Array<Movie>;
  voteClick: VoteClick;
}

export const MovieTable:React.FC<MovieTableProps> = ({movieInfo, voteClick}) => {
 
  return (
      <Table>
        <Caption>Vote for your favorite Star Wars movie:</Caption>
          <THead>
          <Tr>
            <Th id="movie-title">Movie Title</Th>
            <Th id="vote">Vote</Th>
            <Th id="total-votes">Total Votes</Th>
          </Tr>
          </THead>
        <TBody>
            {
              movieInfo.map((movie:Movie) => {
                return (
                  <Tr key={movie.title}>
                    <Td headers="movie-title">{movie.title}</Td>
                    <Td headers="vote">
                      <Button 
                        onClick={() => voteClick(movie.title)} 
                        data-testid={`button-${movie.title}`} 
                        aria-label="vote-button"
                      >
                        vote
                      </Button>
                    </Td>
                    <Td headers="total-votes" data-testid={`votes-${movie.title}`} >{movie.votes}</Td>
                  </Tr>
                )
              })
            }
        </TBody>
      </Table>
  )
}