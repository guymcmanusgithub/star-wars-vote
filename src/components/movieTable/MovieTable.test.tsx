import React from 'react'
import { MovieTable } from './MovieTable'
import { MovieTableProps } from './MovieTable'
import { cleansedData } from '../../mockTestData'
import { render, fireEvent, screen } from '@testing-library/react'

const props:MovieTableProps = {
  movieInfo: cleansedData,
  voteClick: jest.fn()
}

describe('<MovieTable/> component', () => {
 
  it('renders <MovieTable/> component with props and compares to snapshot', () => {
    const wrapper = render(<MovieTable {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
  
  it('renders a vote button for Attack of the Clones', () => {
    render(<MovieTable {...props} />)
    const button = screen.getByTestId('button-Attack of the Clones')
    expect(fireEvent.click(button)).toBeTruthy()
  })

  it('renders a vote button for each of the six movies',  () => {
    render(<MovieTable {...props} />)
    const button = screen.getAllByText("vote")
    expect(button).toHaveLength(6)
  })

})