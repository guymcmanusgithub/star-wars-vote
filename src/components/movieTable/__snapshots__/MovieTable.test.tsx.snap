// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieTable/> component renders <MovieTable/> component with props and compares to snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="sc-bdfBwQ kSOtiz"
      >
        <caption
          class="sc-gsTCUz iNWYbO"
        >
          Vote for your favorite Star Wars movie:
        </caption>
        <thead
          class="sc-dlfnbm"
        >
          <tr
            class="sc-eCssSg gSspkk"
          >
            <th
              class="sc-jSgupP kXXhUQ"
              id="movie-title"
            >
              Movie Title
            </th>
            <th
              class="sc-jSgupP kXXhUQ"
              id="vote"
            >
              Vote
            </th>
            <th
              class="sc-jSgupP kXXhUQ"
              id="total-votes"
            >
              Total Votes
            </th>
          </tr>
        </thead>
        <tbody
          class="sc-hKgILt"
        >
          <tr
            class="sc-eCssSg gSspkk"
          >
            <td
              class="sc-gKsewC gbAxPW"
              headers="movie-title"
            >
              A New Hope
            </td>
            <td
              class="sc-gKsewC gbAxPW"
              headers="vote"
            >
              <button
                aria-label="vote-button"
                class="sc-iBPRYJ cdPLuh"
                data-testid="button-A New Hope"
              >
                vote
              </button>
            </td>
            <td
              class="sc-gKsewC gbAxPW"
              data-testid="votes-A New Hope"
              headers="total-votes"
            >
              0
            </td>
          </tr>
          <tr
            class="sc-eCssSg gSspkk"
          >
            <td
              class="sc-gKsewC gbAxPW"
              headers="movie-title"
            >
              The Empire Strikes Back
            </td>
            <td
              class="sc-gKsewC gbAxPW"
              headers="vote"
            >
              <button
                aria-label="vote-button"
                class="sc-iBPRYJ cdPLuh"
                data-testid="button-The Empire Strikes Back"
              >
                vote
              </button>
            </td>
            <td
              class="sc-gKsewC gbAxPW"
              data-testid="votes-The Empire Strikes Back"
              headers="total-votes"
            >
              3
            </td>
          </tr>
          <tr
            class="sc-eCssSg gSspkk"
          >
            <td
              class="sc-gKsewC gbAxPW"
              headers="movie-title"
            >
              Return of the Jedi
            </td>
            <td
              class="sc-gKsewC gbAxPW"
              headers="vote"
            >
              <button
                aria-label="vote-button"
                class="sc-iBPRYJ cdPLuh"
                data-testid="button-Return of the Jedi"
              >
                vote
              </button>
            </td>
            <td
              class="sc-gKsewC gbAxPW"
              data-testid="votes-Return of the Jedi"
              headers="total-votes"
            >
              0
            </td>
          </tr>
          <tr
            class="sc-eCssSg gSspkk"
          >
            <td
              class="sc-gKsewC gbAxPW"
              headers="movie-title"
            >
              The Phantom Menace
            </td>
            <td
              class="sc-gKsewC gbAxPW"
              headers="vote"
            >
              <button
                aria-label="vote-button"
                class="sc-iBPRYJ cdPLuh"
                data-testid="button-The Phantom Menace"
              >
                vote
              </button>
            </td>
            <td
              class="sc-gKsewC gbAxPW"
              data-testid="votes-The Phantom Menace"
              headers="total-votes"
            >
              6
            </td>
          </tr>
          <tr
            class="sc-eCssSg gSspkk"
          >
            <td
              class="sc-gKsewC gbAxPW"
              headers="movie-title"
            >
              Attack of the Clones
            </td>
            <td
              class="sc-gKsewC gbAxPW"
              headers="vote"
            >
              <button
                aria-label="vote-button"
                class="sc-iBPRYJ cdPLuh"
                data-testid="button-Attack of the Clones"
              >
                vote
              </button>
            </td>
            <td
              class="sc-gKsewC gbAxPW"
              data-testid="votes-Attack of the Clones"
              headers="total-votes"
            >
              0
            </td>
          </tr>
          <tr
            class="sc-eCssSg gSspkk"
          >
            <td
              class="sc-gKsewC gbAxPW"
              headers="movie-title"
            >
              Revenge of the Sith
            </td>
            <td
              class="sc-gKsewC gbAxPW"
              headers="vote"
            >
              <button
                aria-label="vote-button"
                class="sc-iBPRYJ cdPLuh"
                data-testid="button-Revenge of the Sith"
              >
                vote
              </button>
            </td>
            <td
              class="sc-gKsewC gbAxPW"
              data-testid="votes-Revenge of the Sith"
              headers="total-votes"
            >
              2
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="sc-bdfBwQ kSOtiz"
    >
      <caption
        class="sc-gsTCUz iNWYbO"
      >
        Vote for your favorite Star Wars movie:
      </caption>
      <thead
        class="sc-dlfnbm"
      >
        <tr
          class="sc-eCssSg gSspkk"
        >
          <th
            class="sc-jSgupP kXXhUQ"
            id="movie-title"
          >
            Movie Title
          </th>
          <th
            class="sc-jSgupP kXXhUQ"
            id="vote"
          >
            Vote
          </th>
          <th
            class="sc-jSgupP kXXhUQ"
            id="total-votes"
          >
            Total Votes
          </th>
        </tr>
      </thead>
      <tbody
        class="sc-hKgILt"
      >
        <tr
          class="sc-eCssSg gSspkk"
        >
          <td
            class="sc-gKsewC gbAxPW"
            headers="movie-title"
          >
            A New Hope
          </td>
          <td
            class="sc-gKsewC gbAxPW"
            headers="vote"
          >
            <button
              aria-label="vote-button"
              class="sc-iBPRYJ cdPLuh"
              data-testid="button-A New Hope"
            >
              vote
            </button>
          </td>
          <td
            class="sc-gKsewC gbAxPW"
            data-testid="votes-A New Hope"
            headers="total-votes"
          >
            0
          </td>
        </tr>
        <tr
          class="sc-eCssSg gSspkk"
        >
          <td
            class="sc-gKsewC gbAxPW"
            headers="movie-title"
          >
            The Empire Strikes Back
          </td>
          <td
            class="sc-gKsewC gbAxPW"
            headers="vote"
          >
            <button
              aria-label="vote-button"
              class="sc-iBPRYJ cdPLuh"
              data-testid="button-The Empire Strikes Back"
            >
              vote
            </button>
          </td>
          <td
            class="sc-gKsewC gbAxPW"
            data-testid="votes-The Empire Strikes Back"
            headers="total-votes"
          >
            3
          </td>
        </tr>
        <tr
          class="sc-eCssSg gSspkk"
        >
          <td
            class="sc-gKsewC gbAxPW"
            headers="movie-title"
          >
            Return of the Jedi
          </td>
          <td
            class="sc-gKsewC gbAxPW"
            headers="vote"
          >
            <button
              aria-label="vote-button"
              class="sc-iBPRYJ cdPLuh"
              data-testid="button-Return of the Jedi"
            >
              vote
            </button>
          </td>
          <td
            class="sc-gKsewC gbAxPW"
            data-testid="votes-Return of the Jedi"
            headers="total-votes"
          >
            0
          </td>
        </tr>
        <tr
          class="sc-eCssSg gSspkk"
        >
          <td
            class="sc-gKsewC gbAxPW"
            headers="movie-title"
          >
            The Phantom Menace
          </td>
          <td
            class="sc-gKsewC gbAxPW"
            headers="vote"
          >
            <button
              aria-label="vote-button"
              class="sc-iBPRYJ cdPLuh"
              data-testid="button-The Phantom Menace"
            >
              vote
            </button>
          </td>
          <td
            class="sc-gKsewC gbAxPW"
            data-testid="votes-The Phantom Menace"
            headers="total-votes"
          >
            6
          </td>
        </tr>
        <tr
          class="sc-eCssSg gSspkk"
        >
          <td
            class="sc-gKsewC gbAxPW"
            headers="movie-title"
          >
            Attack of the Clones
          </td>
          <td
            class="sc-gKsewC gbAxPW"
            headers="vote"
          >
            <button
              aria-label="vote-button"
              class="sc-iBPRYJ cdPLuh"
              data-testid="button-Attack of the Clones"
            >
              vote
            </button>
          </td>
          <td
            class="sc-gKsewC gbAxPW"
            data-testid="votes-Attack of the Clones"
            headers="total-votes"
          >
            0
          </td>
        </tr>
        <tr
          class="sc-eCssSg gSspkk"
        >
          <td
            class="sc-gKsewC gbAxPW"
            headers="movie-title"
          >
            Revenge of the Sith
          </td>
          <td
            class="sc-gKsewC gbAxPW"
            headers="vote"
          >
            <button
              aria-label="vote-button"
              class="sc-iBPRYJ cdPLuh"
              data-testid="button-Revenge of the Sith"
            >
              vote
            </button>
          </td>
          <td
            class="sc-gKsewC gbAxPW"
            data-testid="votes-Revenge of the Sith"
            headers="total-votes"
          >
            2
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
